src/angularVerticalTree.js

```diff
@@ -85,6 +85,7 @@ angular.module( 'drg.angularVerticalTree', [] )
     const opts = $scope.opts = deepExtend( {
         root : 'Root',
         idProp : undefined,
+        trackByIdProp : undefined,
```

Comment:
I feel like we probably need 2 new options: 
- trackByProp: (string) overrides idProp for the track by
- trackByIndex: (boolean) overrides all the others and forces track by $index

That way we don't lose the performance benefit for current implementations but we also allow tracking by $index to be forced.